File: DiskTudoAdmin.APP/src/app/_models/Produto.ts
import { Pedido } from 'src/app/_models/Pedido';
export class Produto {
    id: number;
    nomeProduto: string;
    descricao: string;
    valor: number;
    qtdade: number;
    imagemURL: string;
    categoriaId?: number;
}

File: DiskTudo.Domain/Produto.cs
using System.Collections.Generic;

namespace DiskTudo.Domain
{
    public class Produto
    {
        public int Id { get; set; }
        public string NomeProduto { get; set; }
        public string Descricao { get; set; }
        public decimal Valor { get; set; }
        public string ImagemURL { get; set; }
        public int? CategoriaId { get; set; }
        public Categoria Categoria { get; }

        public List<ProdutoPedido> ProdutoPedido { get; set; }
    }
}